src/porepy/applications/boundary_conditions/model_boundary_conditions.py

```diff
@@ -0,0 +1,208 @@
+
+
+class BoundaryConditionsEnergyDirNorthSouth(pp.BoundaryConditionMixin):
+    """Boundary conditions for the thermal problem.
+
+    Dirichlet boundary conditions are defined on the north and south boundaries. Some
+    of the default values may be changed directly through attributes of the class.
+
+    Usage: tests for models defining equations for any subset of the thermoporomechanics
+    problem.
+
+    """
+
+    def bc_type_fourier_flux(self, sd: pp.Grid) -> pp.BoundaryCondition:
+        """Boundary condition type for the Fourier heat flux.
+
+        Dirichlet boundary conditions are defined on the north and south boundaries.
+
+        Parameters:
+            sd: Subdomain for which to define boundary conditions.
+
+        Returns:
+            bc: Boundary condition object.
+
+        """
+        domain_sides = self.domain_boundary_sides(sd)
+        # Define boundary condition on faces
+        return pp.BoundaryCondition(sd, domain_sides.north + domain_sides.south, "dir")
+
+    def bc_type_enthalpy_flux(self, sd: pp.Grid) -> pp.BoundaryCondition:
+        """Boundary condition type for the enthalpy.
+
+        Dirichlet boundary conditions are defined on the north and south boundaries.
+
+        Parameters:
+            sd: Subdomain for which to define boundary conditions.
+
+        Returns:
+            bc: Boundary condition object.
+
+        """
+        domain_sides = self.domain_boundary_sides(sd)
+        # Define boundary condition on faces
+        return pp.BoundaryCondition(sd, domain_sides.north + domain_sides.south, "dir")
+
+
+class BoundaryConditionsMechanicsDirNorthSouth(pp.BoundaryConditionMixin):
+    """Boundary conditions for the mechanics with Dirichlet conditions on north and
+    south boundaries.
+
+    """
+
+    params: dict[str, Any]
+    """Model parameters."""
+    solid: pp.SolidConstants
+    """Solid parameters."""
+    fluid: pp.FluidConstants
+    """Fluid parameters."""
+    nd: int
+    """Number of dimensions."""
+
+    def bc_type_mechanics(self, sd):
+        """Boundary condition type for mechanics.
+
+        Dirichlet boundary conditions are defined on the north and south boundaries.
+
+        Parameters:
+            sd: Subdomain for which to define boundary conditions.
+
+        Returns:
+            bc: Boundary condition object.
+
+        """
+        domain_sides = self.domain_boundary_sides(sd)
+        bc = pp.BoundaryConditionVectorial(
+            sd, domain_sides.north + domain_sides.south, "dir"
+        )
+        bc.internal_to_dirichlet(sd)
+        return bc
+
+    def bc_values_displacement(self, boundary_grid: pp.BoundaryGrid) -> np.ndarray:
+        """Boundary values for the mechanics problem as a numpy array.
+
+        Extracted from below method to facilitate time dependent boundary conditions.
+        Values for north and south faces are set to zero unless otherwise specified
+        through attributes ux_north, uy_north, ux_south, uy_south.
```

Comment:
Strictly speaking they are not attributes, but items in the model's parameter dictionary. Should we call them 'model parameters'? I'm not fully convinced.